database/qb/insert.go

```diff
@@ -27,9 +27,8 @@ func (q *InsertQuery) Values(values ...interface{}) *InsertQuery {
 }
 
 // OnDuplicate update these fields / values
-func (q *InsertQuery) OnDuplicate(fields []TableField, values ...interface{}) *InsertQuery {
+func (q *InsertQuery) OnDuplicate(fields []TableField) *InsertQuery {
 	q.onDuplicate = append(q.onDuplicate, fields...)
```

Comment:
should we be filtering the primary key and any non-writable fields here?

Reply:
We can't here because the record object is the one that knows which fields are readable or writable, not the Table or its TableFields. The onus is on the caller

Reply:
fair enough